### PR TITLE
`SubscriptionManager` class, bugfixes

### PR DESCRIPTION
**Pull request by droserasprout**
Improvements:

* Class that holds state of TzKT SignalR subscriptions
* Formatted logging of index messages

Bugfixes:

* Fix displaying help messages on known errors
* Fix missing `ctx.logger` in job handlers
* Fix incorrect job scheduler configuration

### CODE REVIEW

**Comment by droserasprout** [comment]
[Tezos profiles indexer](https://trello.com/c/z3QDMWX7/176-tezos-profiles-indexer)

**Comment by droserasprout** [comment]
[Help messages not printed](https://trello.com/c/oNMhB16L/202-help-messages-not-printed)

**Comment by droserasprout** [comment]
[Logs: index name](https://trello.com/c/49mgHkba/162-logs-index-name)

File: src/dipdup/config.py [hunk 1]
```diff
@@ -545,6 +568,7 @@ class OperationIndexConfig(IndexConfig):
     contracts: Optional[List[Union[str, ContractConfig]]] = None
 
     stateless: bool = False
+    subscribe_by_entrypoints: bool = False
```
Comment (m-kus) [inline src/dipdup/config.py:571]: More generic approach:
```
subscribe: all | by_entrypoints | by_code (future) | other
```

File: src/dipdup/config.py [hunk 2]
```diff
@@ -557,6 +581,15 @@ def contract_configs(self) -> List[ContractConfig]:
                 raise RuntimeError('Config is not initialized')
         return cast(List[ContractConfig], self.contracts)
 
+    @property
+    def entrypoints(self) -> Set[str]:
+        entrypoints = set()
+        for handler in self.handlers:
+            for pattern in handler.pattern:
+                if isinstance(pattern, OperationHandlerTransactionPatternConfig) and pattern.entrypoint:
+                    entrypoints.add(pattern.entrypoint)
```
Comment (m-kus) [inline src/dipdup/config.py:590]: This is not entirely correct, consider the following example:
```
index:
  contracts:
    - my_contract
  handlers:
    - pattern:
        - destination: my_contract
          entrypoint: tezToToken
        - destination: token_contract
          entrypoint: transfer
```
Expected ep set: `[ 'tezToToken' ]`
Actual: `[ 'tezToToken', 'transfer' ]`

File: src/dipdup/datasources/datasource.py [hunk 3]
```diff
@@ -71,3 +83,52 @@ def emit_rollback(self, from_level: int, to_level: int) -> None:
 
     def emit_head(self, block: HeadBlockData) -> None:
         super().emit(EventType.head, datasource=self, block=block)
+
+
+@dataclass
+class Subscriptions:
+    address_transactions: Set[str] = Field(default_factory=set)
```
Comment (m-kus) [inline src/dipdup/datasources/datasource.py:90]: Should be grouped by channel instead, for each channel there's a different algorithm for calculating a subscription hash
1. head -> single one
2. blocks -> single one
3. operations -> type! + address? + entrypoint?  # we basically flatten original subscription params
4. big_maps -> ptr? + tag? + contract? + path?

Then the algorithm of adding a new subscriptions will be the following:
1. Flatten subscription args (if applicable)
2. For each subset check if there's an active subscription
3. If not — create a new one

**Review by m-kus — APPROVED** [review]

**Pull request merged**